src/Elliptic/Systems/Xcts/Geometry.hpp

```diff
@@ -0,0 +1,9 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+namespace Xcts {
+/// \brief Euclidean or non-Euclidean background geometry
+enum class Geometry { Euclidean, NonEuclidean };
```

Comment:
Not sure, the actual code duplication here is minimal since the enums mostly exist for documentation and to write `FlatCartesian`/`Curved` instead of booleans. I think it might be better to keep the enum and its documentation specific to the system instead of trying to generalize it too much. Either could work though 🤷‍♂️